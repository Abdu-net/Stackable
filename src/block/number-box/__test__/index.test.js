import { edit, save } from '../index'

describe( 'Number Box', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				name: 'Registration',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '1',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( '2 column block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				numberBoxTwo: '02',
				name: 'Registration',
				nameTwo: 'Waiting Period',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyTwo: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '2',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( '3 column block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				numberBoxTwo: '02',
				numberBoxThree: '03',
				name: 'Registration',
				nameTwo: 'Waiting Period',
				nameThree: 'Delivery',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyTwo: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyThree: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '3',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				name: 'Registration',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '1',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( '2 column block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				numberBoxTwo: '02',
				name: 'Registration',
				nameTwo: 'Waiting Period',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyTwo: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '2',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( '3 column block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				numberBox: '01',
				numberBoxTwo: '02',
				numberBoxThree: '03',
				name: 'Registration',
				nameTwo: 'Waiting Period',
				nameThree: 'Delivery',
				body: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyTwo: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				bodyThree: 'This is just a sample write-up, but you can check out more info on Gutenberg on the WP repository.',
				numberBoxColor: '#444444',
				nameColor: '#444444',
				bodyTextColor: '#444444',
				numberBGColor: '#dddddd',
				columns: '3',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
