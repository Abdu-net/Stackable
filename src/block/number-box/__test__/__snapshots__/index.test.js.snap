// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number Box block saved block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-number-box ugb-number-box ugb-number-box--v2 ugb-number-box--columns-1 ugb-number-box--design-design className"
  >
    <div
      className="ugb-number-box__item ugb--shadow-1"
      style={
        Object {
          "backgroundColor": "#123456",
          "borderRadius": 6,
        }
      }
    >
      <Unknown
        className="ugb-number-box__number"
        format="string"
        style={
          Object {
            "backgroundColor": "#123456",
            "color": "#123456",
          }
        }
        tagName="span"
        value="num1"
      />
      <div
        className="ugb-number-box__content"
      >
        <Unknown
          className="ugb-number-box__title"
          format="string"
          style={
            Object {
              "color": "#123456",
            }
          }
          tagName="h4"
          value="title1"
        />
        <Unknown
          className="ugb-number-box__description"
          format="string"
          style={
            Object {
              "color": "#123456",
            }
          }
          tagName="p"
          value="description1"
        />
      </div>
    </div>
  </div>
</BlockContentProvider>
`;

exports[`Number Box block saved default block HTML shouldn't change 1`] = `
<BlockContentProvider
  innerBlocks={Array []}
>
  <div
    className="wp-block-ugb-number-box ugb-number-box ugb-number-box--v2 ugb-number-box--columns-3"
  >
    <div
      className="ugb-number-box__item"
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": undefined,
        }
      }
    >
      <Unknown
        className="ugb-number-box__number"
        format="string"
        style={
          Object {
            "backgroundColor": undefined,
            "color": undefined,
          }
        }
        tagName="span"
        value="01"
      />
      <div
        className="ugb-number-box__content"
      >
        <Unknown
          className="ugb-number-box__title"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="h4"
          value="Title"
        />
        <Unknown
          className="ugb-number-box__description"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Description for this block. Use this space for describing your block. Any text will do."
        />
      </div>
    </div>
    <div
      className="ugb-number-box__item"
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": undefined,
        }
      }
    >
      <Unknown
        className="ugb-number-box__number"
        format="string"
        style={
          Object {
            "backgroundColor": undefined,
            "color": undefined,
          }
        }
        tagName="span"
        value="02"
      />
      <div
        className="ugb-number-box__content"
      >
        <Unknown
          className="ugb-number-box__title"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="h4"
          value="Title"
        />
        <Unknown
          className="ugb-number-box__description"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Description for this block. Use this space for describing your block. Any text will do."
        />
      </div>
    </div>
    <div
      className="ugb-number-box__item"
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": undefined,
        }
      }
    >
      <Unknown
        className="ugb-number-box__number"
        format="string"
        style={
          Object {
            "backgroundColor": undefined,
            "color": undefined,
          }
        }
        tagName="span"
        value="03"
      />
      <div
        className="ugb-number-box__content"
      >
        <Unknown
          className="ugb-number-box__title"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="h4"
          value="Title"
        />
        <Unknown
          className="ugb-number-box__description"
          format="string"
          style={
            Object {
              "color": undefined,
            }
          }
          tagName="p"
          value="Description for this block. Use this space for describing your block. Any text will do."
        />
      </div>
    </div>
  </div>
</BlockContentProvider>
`;
